File: Projects/j2me-web-emulator/emulator/canvas.js
export class J2MECanvas{

    constructor(screen){

        this.screen = screen

        this.graphics = null

    }

    paint(graphics){

        // game sẽ override
    }

    repaint(){

        this.paint(this.graphics)

    }

}

export class Canvas{

    paint(g){}

    keyPressed(key){}

    keyReleased(key){}

}

File: Projects/j2me-web-emulator/samples/simple-game/stubs/javax/microedition/lcdui/Canvas.java
package javax.microedition.lcdui;

public abstract class Canvas extends Displayable {
    public static final int UP = -1;
    public static final int DOWN = -2;
    public static final int LEFT = -3;
    public static final int RIGHT = -4;
    public static final int FIRE = -5;

    public Canvas() {
    }

    protected abstract void paint(Graphics g);

    protected void keyPressed(int keyCode) {
    }

    protected void keyReleased(int keyCode) {
    }

    public void repaint() {
    }
}
